agent/hcp/telemetry_provider.go

```diff
@@ -43,57 +44,54 @@ type hcpProviderImpl struct {
 // dynamicConfig is a set of configurable settings for metrics collection, processing and export.
 // fields MUST be exported to compute hash for equals method.
 type dynamicConfig struct {
-	Endpoint *url.URL
-	Labels   map[string]string
-	Filters  *regexp.Regexp
+	disabled bool
```

Comment:
Lowercased these as we don't do the hashing anymore. Missed this in the previous PR.